### DISCUSSION TITLE
Why are Height, Width, Length and Weight integer instead of type decimal?

### DISCUSSION THREAD

**Discussion by teleputer** [medusajs/medusa#3794] [Feature Requests]
Why are the fields of Height, Width, Length and Weight integer instead of decimal?  Consumers and even 3rd party systems need to know the exact measurements and weights of products.   Could you please explain?

**Comment by ivanskodje** [comment]
I agree with this, this should absolutely be allowed. It seem very arbitrary to force upon integers here. 
If there is a good reason to keep it integers, it should probably be documented as well.

That being said, I have found some workarounds besides using metadata for sizes.
If we are setting the unit of measurement to something very low, it can work as-is.

For example with the metric system, it would be enough to set "mm" instead of "cm". For weight, instead of "kg", we would use "g". If you want to display these in "kg" or "cm", you would then handle that logic in the frontend with conversion. 

I assume you can do the same with the imperial system of measurement.

**Comment by moheldesoqy** [comment]
is there any updates on this? the fact dimensions are in integer are kinda a big deal for certain products

**Reply by ivanskodje** [reply]
Until this feature is implemented, you can simplify measurements by using smaller units. For instance, use inches instead of feet or centimeters instead of meters. This change allows you to use whole numbers like 1500 instead of decimal values like 1.5, which can simplify calculations and conversions in the frontend.

**Comment by t-lock** [comment]
Theoretically, could we just migrate the underlying postgres table column to `float4`? In typescript/javascript floats and ints are both just "numbers", so I can't see this breaking anything in the code per se, but I would be very hesitant to modify core Medusa tables. 🤔

**Comment by JohnGemstone** [comment]
I just migrated over 2000 products and most of the weights are wrong because of the conversion to int.
I wonder what the motivation behind this was

**Comment by Ludevsoftware** [comment]
Any update about this? It doesn't make sense to be int. Weight is a continuous quantity. An object can have any mass value, including fractions.

**Comment by jocenunes** [comment]
Any update about this?

**Comment by alexc-tc** [comment]
Any update about this?

**Comment by pkrakas** [comment]
@alexc-tc Looks like it is float already in latest medusa versions

[https://github.com/medusajs/medusa/blob/v2.15.5/packages/modules/product/src/models/product.ts#L47C5-L62C37](https://github.com/medusajs/medusa/blob/v2.15.5/packages/modules/product/src/models/product.ts#L47C5-L62C37)
